### PR TITLE
Update list order interface, change limit behaviour

### PR DESCRIPTION
**Pull request by kevinlacaille**
Closing #380.

### CODE REVIEW

**Comment by kevinlacaille** [comment]
To test the new functionality, we need to still update:
- `tests.integration.test_orders_cli.test_cli_orders_list_limit()`
- `tests.integration.test_orders_api.test_list_orders_limit()` 

New functionality:
- No limit given -> only list up to 100 orders
- Limit set to 0 -> list all orders (limitless)

File: planet/cli/orders.py [hunk 1]
```diff
@@ -57,18 +57,21 @@ def orders(ctx, base_url):
               help='Filter orders to given state.',
               type=click.Choice(planet.clients.orders.ORDER_STATE_SEQUENCE,
                                 case_sensitive=False))
-@click.option('-l',
-              '--limit',
-              help='Filter orders to given limit.',
+@click.option('--limit',
+              help='Maximum number of results to return. Default is 100. A '
+                   'value of 0 means no maximum.',
```
Comment (sgillies) [inline planet/cli/orders.py:62]:
```suggestion
                   'value of 0 means no limit.',
```

File: planet/cli/orders.py [hunk 2]
```diff
@@ -57,18 +57,21 @@ def orders(ctx, base_url):
...
               default=None,
               type=int)
 @pretty
 async def list(ctx, state, limit, pretty):
-    '''List orders'''
+    '''List orders
+
+    This command outputs a sequence of the returned order descriptions.
+    If --pretty is specified, each order description is pretty-printed.
```
Comment (sgillies) [inline planet/cli/orders.py:70]: The text in line 70 could go into the help string for the --pretty option?

Comment (sgillies) [inline planet/cli/orders.py:70]:
```suggestion
    This command prints a sequence of the returned order descriptions, optionally pretty-printed.
```

File: planet/clients/orders.py [hunk 3]
```diff
@@ -422,6 +422,13 @@ async def list_orders(self, state: str = None, limit: int = None):
         """
         url = self._orders_url()
 
+        # Set default limit to 100
+        if limit is None:
+            limit = 100
+        # Set no limit
+        elif limit == 0:
+            limit = None
```
Comment (sgillies) [inline planet/clients/orders.py:430]: If we change the function definition to `list_orders(self, state: str = None, limit: int = 100)` then we can delete lines 425-427 and avoid the limit/None switcheroo.

File: tests/integration/test_orders_api.py [hunk 4]
```diff
@@ -149,26 +149,36 @@ async def test_list_orders_state_invalid_state(session):
...
 @respx.mock
 @pytest.mark.asyncio
-async def test_list_orders_limit(order_descriptions, session):
+@pytest.mark.parametrize("limit,limitted_list_length", [(None, 100), (0, 102),
+                                                        (1, 1)])
+async def test_list_orders_limit(order_descriptions,
+                                 session,
+                                 limit,
+                                 limitted_list_length):
     nono_page_url = TEST_ORDERS_URL + '?page_marker=OhNoNo'
```
Comment (sgillies) [inline tests/integration/test_orders_api.py:158]: @kevinlacaille I found the problem! The "next" link evaluates to `True` on all the result pages, so we have infinitely many pages and the client never breaks out of the paging loop.

```suggestion
    nono_page_url = None
```

Comment (sgillies) [reply tests/integration/test_orders_api.py:158]: If you really want to see what's going on, try replacing some of your test code with this:

```python
    cl = OrdersClient(session, base_url=TEST_URL)

    # import pdb; pdb.set_trace()
    orders = await cl.list_orders(limit=limit)

    async def iter_orders(orders):
        i = 1
        # import pdb; pdb.set_trace()
        async for page in orders._get_pages():
            for item in page["orders"]:
                yield item
                i += 1
                print(f"item {i}")
                if (orders.limit and i > orders.limit) or i > 200:
                    print(f"Reached limit {i}")
                    break
            else:
                continue
            break

    assert len([o async for o in iter_orders(orders)]) == limitted_list_length
```

File: planet/cli/orders.py (hunk 2 above)
Comment (kevinlacaille) [reply planet/cli/orders.py]: Yes, good call!

File: planet/clients/orders.py (hunk 3 above)
Comment (kevinlacaille) [reply planet/clients/orders.py:430]: Jeez ya, that's way better 🥇

File: tests/integration/test_orders_api.py [hunk 5]
```diff
@@ -149,26 +149,36 @@ async def test_list_orders_state_invalid_state(session):
 
 @respx.mock
 @pytest.mark.asyncio
-async def test_list_orders_limit(order_descriptions, session):
-    nono_page_url = TEST_ORDERS_URL + '?page_marker=OhNoNo'
+@pytest.mark.parametrize("limit,limitted_list_length", [(None, 100), (0, 102),
```
Comment (sgillies) [inline tests/integration/test_orders_api.py:152]: Nit: limited instead of limitted.

**Review by sgillies — APPROVED** [review]
Great work @kevinlacaille ! I made two suggestions.

File: planet/cli/orders.py [hunk 6]
```diff
@@ -57,18 +57,21 @@ def orders(ctx, base_url):
               help='Filter orders to given state.',
               type=click.Choice(planet.clients.orders.ORDER_STATE_SEQUENCE,
                                 case_sensitive=False))
-@click.option('-l',
-              '--limit',
-              help='Filter orders to given limit.',
+@click.option('--limit',
+              help='Maximum number of results to return. Default is 100. A '
+              'value of 0 means no limit.',
               default=None,
```
[the remark this answers is not in the document]
Comment (kevinlacaille) [reply planet/cli/orders.py]: Oh whoopsie, nice find!

File: tests/integration/test_orders_api.py (hunk 5 above)
Comment (kevinlacaille) [reply tests/integration/test_orders_api.py:152]: IDEs need to also have spell check lol - good find, thanks!

**Pull request merged**